### ISSUE TITLE
Few shot learning for Document AI

### ISSUE THREAD

**Issue by SimJeg** [microsoft/i-Code#12]
Hello,

I am working on a practical use-case of Document understanding and wondering if I could leverage UDOP. The goal is to extract key informations from the document (in fields or tables). The trick is that I only have a few training samples (<50) and I don't think VQA would apply as these informations are very specific and not always associated with a clear question.

Here are the 2 options I have in mind : 

- finetuning model. But would 50 sample be enough for UDOP ? How should I deal with tables ? (which don't really look like tables but rather a list without printed rows and columns, as on many receipts)
- leverage a foundation model to perform few shot learning (as in GPT3). Are there text + layout foundation models out there that would work for this ? Or should I do prompt engineering with GPT3, Flan-T5, OPT or equivalent models ? 

I am interested to get your insights for both english-data... and non english (but latin) data,

Many thanks for your inputs, but also for your work, it is very useful to get open sourced models as yours :) 
Simon

**Comment by sandorkonya** [comment]
Hi Simon,

i am working on a similar task, with very few (< 30) examples.
Did not try UDOP yet but I had very promising results with [PAddle NLP's UIE](https://github.com/PaddlePaddle/PaddleNLP/blob/develop/applications/information_extraction/document/README_en.md)

I can send you the colab that is working with their own dataset if you want - maybe saves a bit time =)

**Comment by SimJeg** [comment]
Thanks for the quick reply ! I am interested to get a colab, I'll send you a PM

**Comment by ziyi-yang** [comment]
Hi @SimJeg, thanks for your interest in our work. This is an interesting and practical application case. Few-shot learning with UDOP is always in our discussion. Based on few-shot learning experiments we've conducted (results not included in the paper yet), it's possible you can have reasonable performance with 50 examples: UDOP indeed has some few-sot learning capability.

For UDOP, I'd recommend finetuning (we're expediting the release of pretrained weights). If your task mostly depends on the text modality input, I'd recommend trying GPT. You can try encoding the layout modality as textual input to GPT.

**Comment by SimJeg** [comment]
Thanks for your insight @ziyi-yang  :) For few-shot I would love to have some GPT-layout-ready model to do some prompt-based learning...

Are you working on multi-lingual versions too ? My use case cover multiple languages, so UDOP might not help

Also how would you deal with tables with UDOP ? Existing benchmarks are too easy I think, (see for instance the [VRDU](https://arxiv.org/pdf/2211.15421.pdf) paper by Google on this topic). The goal is to predict nested entities (i.e. extract D fields from N lines). 

Maybe it's a lot of questions for a single issue 😓

**Comment by ziyi-yang** [comment]
Sry for the late reply, had been occupied with paper submissions.

We just tested UDOP on the few-shot learning setting. The accuracy on FUNSD is 83.11 with finetuning on 10 examples (acc. is 91.62 full finetuning on 199 examples). However, UDOP can't do in context learning for few-shot.

Currently we don't have plan for the multilingual version.

We don't have special handling for tables.

**Comment by munish0838** [comment]
> Sry for the late reply, had been occupied with paper submissions.
> 
> We just tested UDOP on the few-shot learning setting. The accuracy on FUNSD is 83.11 with finetuning on 10 examples (acc. is 91.62 full finetuning on 199 examples). However, UDOP can't do in context learning for few-shot.
> 
> Currently we don't have plan for the multilingual version.
> 
> We don't have special handling for tables.

Hi @ziyi-yang could you please share the code for using the model on FUNSD dataset?
